src/com/lms/model/Branch.java

```diff
@@ -33,35 +35,19 @@ public int getId() {
 
 	@Override
 	public int hashCode() {
-		final int prime = 31;
-		int result = 1;
-		result = prime * result + ((address == null) ? 0 : address.hashCode());
-		result = prime * result + id;
-		result = prime * result + ((name == null) ? 0 : name.hashCode());
-		return result;
+		return id;
 	}
 
 	@Override
-	public boolean equals(Object obj) {
```

Comment:
it would be better if you override equals() using only the Id of the object without the other fields !

Reply:
Yes, which Jonathan corrected. I updated this branch with those changes, so we should be good now.